packages/@aws-cdk/aws-rds/test/test.instance.ts

```diff
@@ -1138,4 +1138,64 @@ export = {
       test.done();
     },
   },
+
+  'fromGeneratedPassword'(test: Test) {
+    // WHEN
+    const db1 = new rds.DatabaseInstance(stack, 'Database1', {
+      engine: rds.DatabaseInstanceEngine.postgres({ version: rds.PostgresEngineVersion.VER_12_3 }),
+      vpc,
+      credentials: rds.Credentials.fromGeneratedPassword('postgres'),
+    });
+
+    const db2 = new rds.DatabaseInstance(stack, 'Database2', {
+      engine: rds.DatabaseInstanceEngine.postgres({ version: rds.PostgresEngineVersion.VER_12_3 }),
+      vpc,
+      credentials: rds.Credentials.fromGeneratedPassword('postgres', {
+        excludeCharacters: "<>?!'/@\"\\", // different characters set
+      }),
+    });
+
+    // THEN
+    expect(stack).to(haveResource('AWS::RDS::DBInstance', {
+      MasterUsername: 'postgres', // username is a string
+      MasterUserPassword: {
+        'Fn::Join': [
+          '',
+          [
+            '{{resolve:secretsmanager:',
+            {
+              Ref: 'DefaultDatabase1Secretd0794ea66c2e84ff83bd09c0bced0333', // logic id is a hash
+            },
+            ':SecretString:password::}}',
+          ],
+        ],
+      },
+    }));
+
+    // Check that the logical id of the second secret is different
+    const db1Secret = db1.node.tryFindChild('Secret') as rds.DatabaseSecret;
+    const cfnSecret1 = db1Secret.node.defaultChild as cdk.CfnResource;
+    const db2Secret = db2.node.tryFindChild('Secret') as rds.DatabaseSecret;
+    const cfnSecret2 = db2Secret.node.defaultChild as cdk.CfnResource;
+    test.notEqual(cfnSecret1.logicalId, cfnSecret2.logicalId);
```

Comment:
`logicalIds` are tokens, so you need to resolve them first before you do the comparison (try it be attempting to make this test fail by changing the hashing algorithm, it shouldn't work - this assertion should always pass).

Reply:
of course! moved these kind of assertions to `test.database-secret.ts` where it actually belongs.